src/vs/workbench/contrib/search/browser/searchModel.ts

```diff
@@ -90,13 +90,12 @@ export class Match {
 	}
 
 	@memoize
-	preview(): { before: string; inside: string; after: string } {
-		let before = this._oneLinePreviewText.substring(0, this._rangeInPreviewText.startColumn - 1),
-			inside = this.getMatchString(),
-			after = this._oneLinePreviewText.substring(this._rangeInPreviewText.endColumn - 1);
+	preview(): { before: string; fullBefore: string; inside: string; after: string } {
+		const fullBefore = this._oneLinePreviewText.substring(0, this._rangeInPreviewText.startColumn - 1),
+			before = lcut(fullBefore, 26, '…');
 
-		before = lcut(before, 26);
-		before = before.trimStart();
```

Comment:
We still need `trimStart()` for cases where the string's length is less than 26.

Reply:
@andreamah yes, quite right, I missed that. 

However, looking at this, I've improved my understanding of `lcut()` - it seems that `n` is not a maximum but a target, given a string that's longer than `n`:

  - with whitespace characters before, it can return less than `n` characters,
  - but given non-whitespace characters, it can return more. The original tests show that with `assert.strictEqual(strings.lcut('foo bar', 5), 'foo bar');`
 
I'm adding new tests that show when `lcut()` can return more than `n` characters, which is behaviour not affected by my PR. 

Therefore, it feels that `lcut()` could also trim the start of a string that's already shorter. I've implemented your comment that way.

Reply:
I still see a case where an ellipsis is present where it shouldn't be. See below:
<img width="639" alt="image" src="https://github.com/microsoft/vscode/assets/31675041/8ba5d91f-c80f-4aec-885d-68d6848c923d">


I'm assuming that the `...` represents the leading whitespace, which is inconsistent with other cases where we trim whitespace and don't have a `...`.

Reply:
Thank you again, yes, this wasn't handled well. I've committed a fix. The code feels cleaner now, too.